### ISSUE TITLE
Increase in Training Time for TadGAN implemented in TensorFlow 2.x

### ISSUE THREAD

**Issue by lcwong0928** [sintel-dev/Orion#293]
* Orion version: 0.3.1.dev0
* Python version: 3.7.11
* Operating System: macOS

### Description
There is a significant increase in training time per signal between TadGAN implemented in TensorFlow 2.x and TensorFlow 1.x. The main differences between the two environments are the methodology to compute the gradient penalty loss and `tf.GradientTape()` to compile the model. 

### What I Did
The first model (TF 2.0-1) is based on the [Wasserstein GAN (WGAN) with Gradient Penalty (GP)](https://keras.io/examples/generative/wgan_gp/) tutorial that uses `tf.GradientTape()` for the `train_step` and a second-order `tf.GradientTape()` for the gradient penalty loss.  The second model  (TF 2.0-2) compiles the model similar to the TensorFlow 1.x version but still uses `tf.GradientTape()` for the gradient penalty loss.

The following table reports the average training time across all signals (on GPU).
![image](https://user-images.githubusercontent.com/35314385/168848224-ee4ee1e3-3453-4aaa-a4de-7efe8ece3209.png)

**Comment by sarahmish** [comment]
Thank you @lcwong0928 for the analysis!

Would it be possible to make a CPU comparison between TF1.0 and TF2.0-2?

**Comment by lcwong0928** [comment]
Yes, will run a benchmark for the CPU version.
